spring-cloud-gcp-data-firestore/src/main/java/com/google/cloud/spring/data/firestore/repository/query/PartTreeFirestoreQuery.java

```diff
@@ -216,9 +219,18 @@ public Object execute(Object[] parameters) {
 							"Too few parameters are provided for query method: " + getQueryMethod().getName());
 				}
 				Object value = it.next();
-				filter.getFieldFilterBuilder().setField(fieldReference)
-						.setOp(getOperator(part, value))
-						.setValue(this.classMapper.toFirestoreValue(value));
+				StructuredQuery.FieldFilter.Builder queryFilterBuilder = filter.getFieldFilterBuilder()
+						.setField(fieldReference).setOp(getOperator(part, value));
+				if (fieldReference.getFieldPath().equals(FieldPath.documentId().toString())) {
+					FirestoreTemplate firestoreTemplate = (FirestoreTemplate) this.reactiveOperations;
```

Comment:
In real life, `reactiveOperations` will almost always be of type `FirestoreTemplate`. However, that's not guaranteed!
Also, for testing, if we want to mock `reactiveOperations`, this line will fail. 

I would instead suggest adding `buildResourceName()` as a valid interface method on `FirestoreReactiveOperations` (btw, then the javadoc will go on the interface method, and not on implementation). This will make it a proper part of the API, which we can then mock for testing or be required to implement for an alternate implementation.

Reply:
> Also, for testing, if we want to mock reactiveOperations, this line will fail.

Ah I did not think of this, thank you!